main.cc: specify shortname for scheduling groups

tchaikov opened the pull request:
so, for instance, the logging message looks like:
```
INFO  2023-10-24 15:19:37,290 [shard 0:strm] storage_service - entering STARTING mode
```
instead of
```
INFO  2023-10-24 15:19:37,290 [shard 0:stre] storage_service - entering STARTING mode
```

Fixes #15267
Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

scylladb-promoter commented:
#### :red_circle: CI State: ABORTED

:white_check_mark: - Build
:x: - Sanity Tests
:white_check_mark: - Unit Tests
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4346/
- Duration: 2 hr 30 min
- Builder: spider3.cloudius-systems.com

denesb commented on main.cc:
```diff
@@ -965,14 +965,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
-            dbcfg.gossip_scheduling_group = make_sched_group("gossip", 1000);
-            dbcfg.commitlog_scheduling_group = make_sched_group("commitlog", 1000);
+            dbcfg.statement_scheduling_group = make_sched_group("statement", "stmt", 1000);
+            dbcfg.memtable_scheduling_group = make_sched_group("memtable", "mmtb", 1000);
```
`mt`

denesb commented on main.cc:
```diff
@@ -965,14 +965,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
-            dbcfg.commitlog_scheduling_group = make_sched_group("commitlog", 1000);
+            dbcfg.statement_scheduling_group = make_sched_group("statement", "stmt", 1000);
+            dbcfg.memtable_scheduling_group = make_sched_group("memtable", "mmtb", 1000);
+            dbcfg.memtable_to_cache_scheduling_group = make_sched_group("memtable_to_cache", "mmtc", 200);
```
`mt2c`

denesb commented on main.cc:
```diff
@@ -965,14 +965,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
+            dbcfg.statement_scheduling_group = make_sched_group("statement", "stmt", 1000);
+            dbcfg.memtable_scheduling_group = make_sched_group("memtable", "mmtb", 1000);
+            dbcfg.memtable_to_cache_scheduling_group = make_sched_group("memtable_to_cache", "mmtc", 200);
+            dbcfg.gossip_scheduling_group = make_sched_group("gossip", "gosp", 1000);
```
`gms`

denesb commented on main.cc:
```diff
@@ -965,14 +965,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
+            dbcfg.memtable_scheduling_group = make_sched_group("memtable", "mmtb", 1000);
+            dbcfg.memtable_to_cache_scheduling_group = make_sched_group("memtable_to_cache", "mmtc", 200);
+            dbcfg.gossip_scheduling_group = make_sched_group("gossip", "gosp", 1000);
+            dbcfg.commitlog_scheduling_group = make_sched_group("commitlog", "cmlg", 1000);
```
`clog`

tchaikov commented:
> Suggested alternative short names. We already use abbreviations like `mt` for memtable and `gms` for gossip in the code. Best to reuse these in the short names too.

sure. changed accordingly


v2:

- changed the short name according to the suggestions


@denesb hi Botond, thank you for your review and suggestions. updated and repushed.

denesb commented on main.cc:
```diff
@@ -965,14 +965,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
-            dbcfg.commitlog_scheduling_group = make_sched_group("commitlog", 1000);
+            dbcfg.statement_scheduling_group = make_sched_group("statement", "stmt", 1000);
+            dbcfg.memtable_scheduling_group = make_sched_group("memtable", "mt", 1000);
+            dbcfg.memtable_to_cache_scheduling_group = make_sched_group("memtable_to_cache", "mm2c", 200);
```
`mt2c`, to be consistent with the `mt` above (and the abbreviations used in the code already)

tchaikov replied:
@denesb thank you Botond for the explanation! and sorry, i was somehow thinking about the memory subsystem of linux at that moment, but missed the table part!

tchaikov commented:
v2:

- s/mm2c/mt2c/

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - Build
:x: - Sanity Tests
:x: - Unit Tests
#### Failed Tests (3/21010):
- [test_rolling_upgrade_happy_path](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4366/testReport/junit/%28root%29/test_raft_cluster_features/Tests___Unit_Tests___test_rolling_upgrade_happy_path) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade_happy_path)
- [topology_experimental_raft.test_raft_cluster_features.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4366/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___topology_experimental_raft_test_raft_cluster_features_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+topology_experimental_raft.test_raft_cluster_features.debug.2)
- [test_lcs_table_promotion_major_compaction](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4366/testReport/junit/compaction_additional_test/TestLCSSSTablePromotion/Tests___Sanity_Tests___test_lcs_table_promotion_major_compaction) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_lcs_table_promotion_major_compaction)
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4366/
- Duration: 7 hr 32 min
- Builder: i-0c40efea6d918cdf4

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - Build
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4378/
- Duration: 55 min
- Builder: spider5.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - Build
:white_check_mark: - Sanity Tests
:white_check_mark: - Unit Tests
#### Build Details:
- Build URL: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/4394/
- Duration: 2 hr 21 min
- Builder: spider8.cloudius-systems.com
